static/app/views/profiling/profileDetails/utils.ts

```diff
@@ -0,0 +1,93 @@
+import {CallTreeNode} from 'sentry/utils/profiling/callTreeNode';
+import {Profile} from 'sentry/utils/profiling/profile/profile';
+
+export function collectProfileFrames(profile: Profile) {
+  const nodes: CallTreeNode[] = [];
+
+  profile.forEach(
+    node => {
+      if (node.selfWeight > 0) {
+        nodes.push(node);
+      }
+    },
+    () => {}
+  );
+
+  return nodes
+    .sort((a, b) => b.selfWeight - a.selfWeight)
+    .map(node => ({
+      symbol: node.frame.name,
+      file: node.frame.file,
+      image: node.frame.image,
+      thread: profile.threadId,
+      type: node.frame.is_application ? 'application' : 'system',
+      'self weight': node.selfWeight,
+      'total weight': node.totalWeight,
+    }));
+}
+
+export function pluckUniqueValues<T extends Record<string, any>>(
+  collection: T[],
+  key: keyof T
+) {
+  return collection.reduce((acc, record) => {
+    const value = record[key];
+    if (value && !acc.includes(value)) {
+      acc.push(value);
+    }
+    return acc;
+  }, [] as string[]);
+}
+
+export type Row<K extends string = string> = Record<
+  Extract<K, string>,
+  string | number | undefined
+>;
+export interface AggregateColumnConfig<K extends string> {
+  compute: (data: Row<K>[]) => number;
+  key: string;
+}
+export function aggregate<T extends string>(
+  data: Partial<Row<T>>[],
+  groups: Extract<T, string>[],
+  aggregates: AggregateColumnConfig<T>[]
+): Row<T>[] {
+  const groupedData = groupBy(data, groups);
+
+  const rows: Row[] = [];
+  for (const groupedKey in groupedData) {
+    const row = makeRowFromGroupedKey(groupedKey, groups);
+    const groupedValues = groupedData[groupedKey];
+    aggregates.forEach(agg => {
+      row[agg.key] = agg.compute(groupedValues);
+    });
+    rows.push(row);
+  }
+  return rows;
+}
+
+function groupBy<T extends Row>(data: T[], groups: Extract<keyof T, string>[]) {
+  return data.reduce((acc, row) => {
+    const key = getGroupedKey(row, groups);
+    if (!acc[key]) {
+      acc[key] = [];
+    }
+
+    acc[key].push(row);
+    return acc;
+  }, {});
+}
+
+const FIELD_SEPARATOR = String.fromCharCode(31);
```

Comment:
add comment to this, it's a non printable system character used to delimit fields. i've chosen this to avoid any conflicts with any frame data we may receive.

see:
- https://en.wikipedia.org/wiki/Delimiter#ASCII_delimited_text
- https://stackoverflow.com/questions/8695118/what-are-the-file-group-record-unit-separator-control-characters-and-their-usage#:~:text=A%20Unit%20separator%20could%20provide,in%20a%20tab%2Ddelimited%20file.

Reply:
I would add that comment in the code directly. Just link to that wikipedia article is great

Reply:
yeah that was my note to self, to add a comment in the code 😅